File: solo_explore/lib/models/trip_plan.dart
class TripPlan {
  final int id;
  final String title;
  final String? description;
  final String startDate;
  final String endDate;
  final int totalDays;
  final String status;
  final int itemsCount;
  final String createdAt;

  TripPlan({
    required this.id,
    required this.title,
    this.description,
    required this.startDate,
    required this.endDate,
    required this.totalDays,
    required this.status,
    this.itemsCount = 0,
    required this.createdAt,
  });

  factory TripPlan.fromJson(Map<String, dynamic> json) {
    return TripPlan(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      startDate: json['start_date'],
      endDate: json['end_date'],
      totalDays: json['total_days'] ?? 1,
      status: json['status'] ?? 'draft',
      itemsCount: json['items_count'] ?? 0,
      createdAt: json['created_at'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'description': description,
      'start_date': startDate,
      'end_date': endDate,
      'total_days': totalDays,
      'status': status,
      'items_count': itemsCount,
      'created_at': createdAt,
    };
  }
}

File: Solo_Explore-fitur-umar/solo-explore-backend/app/Models/TripPlan.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class TripPlan extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'user_id',
        'name',
        'title',
        'description',
        'start_date',
        'end_date',
        'total_days',
        'budget',
        'status',
        'ai_generated',
        'ai_itinerary',
    ];

    protected $casts = [
        'start_date' => 'date',
        'end_date' => 'date',
        'total_days' => 'integer',
        'budget' => 'decimal:2',
        'ai_generated' => 'boolean',
    ];

    // Relationships
    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function items()
    {
        return $this->hasMany(TripPlanItem::class)->orderBy('day_number')->orderBy('order');
    }

    // Scopes
    public function scopeActive($query)
    {
        return $query->where('status', 'active');
    }

    public function scopeDraft($query)
    {
        return $query->where('status', 'draft');
    }

    // Methods
    public function calculateTotalDays()
    {
        if ($this->start_date && $this->end_date) {
            $this->total_days = $this->start_date->diffInDays($this->end_date) + 1;
            $this->save();
        }
    }
}
